File: java_solutions/647.palindromic-substrings.java
/*
 * @lc app=leetcode id=647 lang=java
 *
 * [647] Palindromic Substrings
 */

record Solution() {
    public int countSubstrings(String s) {
        return java.util.stream.IntStream.range(0, 2 * s.length() - 1).map(i -> (int) java.util.stream.Stream.iterate(new int[]{i / 2, (i + 1) / 2}, b -> b[0] >= 0 && b[1] < s.length() && s.charAt(b[0]) == s.charAt(b[1]), b -> new int[]{b[0] - 1, b[1] + 1}).count()).sum();
    }
}

File: kotlin_solutions/647.palindromic-substrings.kt
/*
 * @lc app=leetcode id=647 lang=kotlin
 *
 * [647] Palindromic Substrings
 */

class Solution { fun countSubstrings(s: String): Int = s.indices.sumOf { i -> (0..minOf(i, s.lastIndex - i)).takeWhile { k -> s[i - k] == s[i + k] }.count() + (0..minOf(i, s.lastIndex - i - 1)).takeWhile { k -> s[i - k] == s[i + k + 1] }.count() } }
